src/main/java/com/radixdlt/crypto/hdwallet/BitcoinJHDKeyPairDerivation.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ *  * (C) Copyright 2020 Radix DLT Ltd
+ *  *
+ *  * Radix DLT Ltd licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except in
+ *  * compliance with the License.  You may obtain a copy of the
+ *  * License at
+ *  *
+ *  *  http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *  * either express or implied.  See the License for the specific
+ *  * language governing permissions and limitations under the License.
+ *
+ */
+
+package com.radixdlt.crypto.hdwallet;
+import com.google.common.annotations.VisibleForTesting;
+import com.radixdlt.SecurityCritical;
+import com.radixdlt.crypto.CryptoException;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.utils.Bytes;
+import org.bitcoinj.crypto.ChildNumber;
+import org.bitcoinj.crypto.DeterministicHierarchy;
+import org.bitcoinj.crypto.DeterministicKey;
+import org.bitcoinj.crypto.HDKeyDerivation;
+import org.bitcoinj.crypto.MnemonicCode;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.bitcoinj.core.Utils.WHITESPACE_SPLITTER;
+
+@SecurityCritical({ SecurityCritical.SecurityKind.KEY_GENERATION })
+public final class BitcoinJHDKeyPairDerivation implements HDKeyPairDerivation {
+
+	/**
+	 * "m/" marks inclusion of private key, whereas "M/" (capital M) marks public key only.
+	 */
+	private static String BIP32_PRIVATE_KEY_BIP44_PREFIX = "m/";
+
+	private static String BIP32_PATH_DELIMITER = "/";
+	private static String BIP32_HARDENED_PATH_STANDARD_MARKER = "'";
+	private static String BIP32_HARDENED_PATH_BITCOINJ_MARKER = "H";
+
+	private static String BIP39_MNEMONIC_NO_PASSPHRASE = "";
```

Comment:
Seems like all of the above should also have `final` keyword

Reply:
made obsolete